### PR TITLE
Update synapse.py to fix timeout bug

### PR DESCRIPTION
**Pull request by karthikkadajji**
---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by uranusjr** [comment]
Can a test be added for this?

File: airflow/providers/microsoft/azure/hooks/synapse.py [hunk 1]
```diff
@@ -179,7 +179,7 @@ def wait_for_job_run_status(
             and job_run_status not in expected_statuses
         ):
             # Check if the job-run duration has exceeded the ``timeout`` configured.
-            if start_time + timeout < time.monotonic():
+            if start_time + timeout > time.monotonic():
```
Comment (hussein-awala) [inline airflow/providers/microsoft/azure/hooks/synapse.py:182]: Let's consider that the srart_time=0 when we start the execution (`start_time = time.monotonic()`) and timeout = 25:
- 1st iteration: (0 + 25) < 1 -> False -> sleep 10 seconds -> check the status
- 2nd iteration: (0 + 25) < 14 -> False -> sleep 10 seconds -> check the status
- 3rd iteration: (0 + 25) < 27 -> True -> raise `AirflowTaskTimeout`

Why do you think that there is a bug in this condition?

**Review by hussein-awala — CHANGES_REQUESTED** [review]
However, the tests were successful in both conditions, which means there is no test coverage for this timeout, if you are interested in contributing to the project I suggest adding a new test for this timeout.

File: airflow/providers/microsoft/azure/hooks/synapse.py (hunk 1 above)
Comment (karthikkadajji) [reply airflow/providers/microsoft/azure/hooks/synapse.py:182]: you are right. It is the missing test case for timeout

**Pull request closed without merging**